doc/syntax.rst

```diff
@@ -1,40 +1,92 @@
-=====================
-Sphinx-Gallery Syntax
-=====================
+.. _python_script_syntax:
 
-For your examples to work you have to make them valid python code.
+=======================================================
+How to structure your Python scripts for Sphinx-Gallery
+=======================================================
 
+This page describes the structure and syntax that can be used in Python scripts
+to generate rendered HTML gallery pages.
 
-Simple examples
-===============
+A simple example
+================
 
-This is the base of a python script of module. The structure follows
-a leading docstring that must follow reStructuredText syntax and a header.
-Then the executable code of your script.
+Sphinx-Gallery expects each Python file to have two things:
+
+1. **A docstring**, written in rST, that defines the
+   header for the example. It must begin by defining an rST title. For example::
+
+    """
+    This is my example script
+    =========================
+
+    This example doesn't do much, it just makes a simple plot
+    """
+2. **Python code**. This can be any valid Python code that you wish. Any
+   Matplotlib or Mayavi images that are generated will be saved to disk, and
+   the rST generated will display these images with the built examples.
 
 For a quick reference have a look at the example
 :ref:`sphx_glr_auto_examples_plot_gallery_version.py`
 
-In this 2 block structure, the website is rendered ordered with
-the docstring, then output images and output code. Finally the
-generating code script.
+.. _embedding_rst:
+
+Embed rST in your example Python files
+======================================
+
+Additionally, you may embed rST syntax within your Python scripts. This will
+be rendered in-line with the Python code and its outputs, similar to how
+Jupyter Notebooks are structured (in fact, Sphinx-Gallery also **creates** a
+Jupyter Notebook for each example that is built).
+
+You can embed rST in your Python
+examples by including a line of ``#`` symbols that spans 79 columns, like
```

Comment:
we officially ask for more than 20 `#`, 79 is just to be nicely compliant with PEP8 row width but is not enforced in any way.